AzSentinel/Public/Get-AzSentinelAlertRule.ps1

```diff
@@ -75,14 +82,19 @@ function Get-AzSentinelAlertRule {
 
         $return = @()
         if ($alertRules.value) {
+            if ($LastModified) {
```

Comment:
Please move the if statement outside the parent if statement (somewhere on line 82). Because now you can have data when you go in the first if statement and if the time filter returns zero the last else at line 153 will still be executed but it will return error's because $alertRules.value is empty.

or make it a complete filter with the return etc property (like rulename, kind etc), so that it doesn't go to the last else

```PowerShell
        if ($null -ne $alertRules.value -and $LastModified) {
            Write-Verbose "Filtering for rules modified after $LastModified"
            $alertRules.value  = $alertRules.value  | Where-Object { $_.properties.lastModifiedUtc -gt $LastModified }
        }
```

Reply:
Good points, thanks. I think it's better to move it out, then it can be used in conjunction with other filters.